### ISSUE TITLE
Spotify Search doesn't work properly in some cases

### ISSUE THREAD

**Issue by vanishrap** [JohnnyCrazy/SpotifyAPI-NET#131]
Hello,
I found out that you need to add HttpUtility.UrlEncode() method when making a request to Spotify Search.
Otherwise it can return wrong results (more results than needed) or no results at all (if building advanced search query). 
I think this issue is caused by special characters in the search query, that need to be encoded before the request.
Have Fun!

BR,
Vlad.

**Comment by JohnnyCrazy** [comment]
Will be fixed in the upcoming patch, forgot it in this release.

Thanks for the report! 👍

**Comment by JohnnyCrazy** [comment]
@vanishrap Can you give me an example of a search query which fails?

**Comment by vanishrap** [comment]
@JohnnyCrazy I'm sorry for delayed response, I don't have clear reproduction scenario.

I've got different results for this track: "spotify:track:13srjlMr6szk11sdFUKE7O"
Here are the scenarios I've tried:

- Search by artist and track name (exact match, i.e. artist:"..." track:"...", etc)
- Search by artist and track, but not using exact match.

I've also tried to include the album name to these queries.
I think that '&' character in the query can cause some strange issues.
Hope this information is useful for you.
